python/openai/openai/src/utils/triton.py

```diff
@@ -0,0 +1,211 @@
+# Copyright 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import os
+import time
+import typing
+from dataclasses import dataclass
+
+import numpy as np
+import tritonserver
+from fastapi import HTTPException
+from src.schemas.openai import CreateChatCompletionRequest, CreateCompletionRequest
+from src.utils.tokenizer import get_tokenizer
+
+
+# TODO: Stricter pydantic validation would be better in future
+@dataclass
+class TritonModelMetadata:
+    # Name used in Triton model repository
+    name: str
+    # Name of backend used by Triton
+    backend: str
+    # Triton model object handle
+    model: tritonserver.Model
+    # Tokenizers used for chat templates
+    tokenizer: typing.Optional[typing.Any]
+    # Time that model was loaded by Triton
+    create_time: int
+    # Conversion format between OpenAI and Triton requests
+    request_convert_fn: typing.Optional[typing.Any]
```

Comment:
nit: maybe `request_converter` - don't see a lot of `fn` suffix -

Reply:
Renamed and type hinted as `typing.Callable` -- didn't go as far as the expected callable input/output types yet